tools/amplify-preview/main.go

```diff
@@ -0,0 +1,136 @@
+)
+
+var (
+	logger = slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{Level: slog.LevelDebug}))
+
+	amplifyAppIDs  = kingpin.Flag("amplify-app-ids", "List of Amplify App IDs").Envar("AMPLIFY_APP_IDS").Required().Strings()
+	gitBranchName  = kingpin.Flag("git-branch-name", "Git branch name").Envar("GIT_BRANCH_NAME").Required().String()
+	createBranches = kingpin.Flag("create-branches",
+		"Defines whether Amplify branches should be created if missing, or just lookup existing ones").Envar("CREATE_BRANCHES").Default("false").Bool()
+	wait = kingpin.Flag("wait",
+		"Wait for pending/running job to complete").Envar("WAIT").Default("false").Bool()
+)
+
+const (
+	jobWaitSleepTime    = 30 * time.Second
+	jobWaitTimeAttempts = 40
+)
+
+func main() {
+	kingpin.Parse()
+	ctx := context.Background()
+
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRetryer(func() aws.Retryer {
+		return retry.AddWithMaxAttempts(retry.NewStandard(), 10)
+	}))
+	if err != nil {
+		logger.Error("failed to load configuration", "error", err)
+		os.Exit(1)
+	}
+
+	amp := AmplifyPreview{
+		client: amplify.NewFromConfig(cfg),
+		appIDs: func() []string {
+			if len(*amplifyAppIDs) == 1 {
+				// kingpin env variables are separated by new lines, and there is no way to change the behavior
+				// https://github.com/alecthomas/kingpin/issues/249
+				return strings.Split((*amplifyAppIDs)[0], ",")
+			}
+			return *amplifyAppIDs
+		}(),
+	}
+
+	// Check if Amplify branch is already connected to one of the Amplify Apps
+	branch, err := amp.FindExistingBranch(ctx, *gitBranchName)
+	if err != nil {
+		if !*createBranches && !errors.Is(err, errNoJobForBranch) {
+			logger.Error("failed to lookup branch", logKeyBranchName, *gitBranchName, "error", err)
+			os.Exit(1)
+		}
+
+		// If branch wasn't found, and branch creation enabled - create new branch
+		branch, err = amp.CreateBranch(ctx, *gitBranchName)
+		if err != nil {
+			logger.Error("failed to create branch", logKeyBranchName, *gitBranchName, "error", err)
+			os.Exit(1)
+		}
+	}
+
+	// check if existing branch was/being deployed already
+	latestJob, activeJob, err := amp.GetLatestAndActiveJobs(ctx, branch)
+	if err != nil {
+		if !*createBranches && !errors.Is(err, errNoJobForBranch) {
+			logger.Error("failed to get amplify job", logKeyBranchName, *gitBranchName, "error", err)
+			os.Exit(1)
+		}
+
+		// if job not found and branch was just created - start new job
+		latestJob, err = amp.StartJob(ctx, branch)
+		if err != nil {
+			logger.Error("failed to start amplify job", logKeyBranchName, *gitBranchName, "error", err)
+			os.Exit(1)
+		}
+	}
+
+	if err := postPreviewURL(ctx, amplifyJobsToMarkdown(branch, latestJob, activeJob)); err != nil {
+		logger.Error("failed to post preview URL", "error", err)
+		os.Exit(1)
+	}
+	logger.Info("Successfully posted PR comment")
+
+	if *wait {
+		for i := 0; !isAmplifyJobCompleted(latestJob) && i < jobWaitTimeAttempts; i++ {
```

Comment:
E.g. from above comment. This would be easier to understand and follow if it said `if wait { DoSomethingElseWithALiterateName }`. Self-documenting code (with short names) is the best kind of code.